keel-core/src/main/java/com/netflix/spinnaker/keel/serialization/PrecisionSqlSerializer.java

```diff
@@ -35,7 +35,7 @@ public PrecisionSqlSerializer() {
         Instant::toEpochMilli,
         Instant::getEpochSecond,
         Instant::getNano,
-        new DateTimeFormatterBuilder().appendInstant(3).toFormatter());
+        new DateTimeFormatterBuilder().appendInstant(6).toFormatter());
```

Comment:
The comment above about using 3 digits could use a tweak.

Reply:
Was waiting for tests to pass ;)  LOOKS Like they're happy with the 6 precision, though datetime(3) MAY truncate it more.  The question there to my mind is... do we want this to go to 3 or 6?

Reply:
What did we have with java 11?

Reply:
Was the point for this to just allow for more precision? If that's the case, we may want to just leave it at 3. Like I said that was the only thing I was unsure of, but more precision also does not hurt, but Spinnaker also doesn't need ot be the most precise service either

Reply:
Yeah mostly trying to match the most precision to prevent any goofy collision issues we MIGHT hit.  I was originally "leave it at 3" but... if 6 works, AND we're consistent that's really the bigger key in my mind.

Reply:
Yea, agreed. I think leaving it at 6 is fine.

Reply:
Maybe I'm getting tripped up in the wording, but isn't there a difference between what java 11 supports, and what keel actually used?  I'd prefer to keep it the same as what we had if possible.

Reply:
SO not quite.  Java 11 supported and delivered 6 digits of precision at MOST.  Java 17 ALLOWED a higher level of precision if the system clock supported it.  Instant.toString was never consistent on the output - it would just output the MOST available.  This led to situations where when doing serialization to JSON, you'd get inconsistent results.  And on an OS with HIGHER precision, you'd go over the bounds that the database supported.  SO the primary impact and root of the problem is when Keel does auto generated fields OFF of the JSON string in date format, causing a failure.  Java was always able to handle it, but databases didn't handle the higher precision UNLESS they were storing things as a string (which most are).  Timestamp fields however would NOT work.

SO the key on this is to serialize at MAX precision for the storage layer.  NOTE:  There are POTENTIAL headaches on tests in other places if we do comparisons of what's written & read on unit tests in a real database.  SQLLite being Java MAY support a higher preceision - but I'd bet it depends on compatibility mode.  SO if we have a test that writes the timestamps using a mysql test container, then tries to do an equals on a read of that timestamp it will NOT work as mysql wouldn't support the full precision that the JVM supports.

Reply:
To note... Orca uses LONG's for persistence of timestamps - so it worked without issue.  It's JUST keel that stores times as Instants and in string format.  Least so far that I found.  That was the original headache with a simple replacement serialization implementation - I had to switch to the time base serializer which is smarter on it's handling.

Reply:
OK, sounds like we're going with 6.  I guess we don't include keel in our release notes, so we're off the hook there :)